src/tracks/SampleTrack.cpp

```diff
@@ -665,8 +1113,148 @@ void SampleTCOView::reverseSample()
 }
 
 
+void SampleTCOView::mouseMoveFadingPos(const QMouseEvent* e)
+{
+	float xpos = static_cast<float>(e->localPos().x());
+	float width = static_cast<float>(this->width());
+	float faderPos = qMin(qMax(0.0f, xpos / width), 1.0f);
+	if (m_leftCorner)
+	{
+		m_tco->m_sampleBuffer->setLeftFader(faderPos);
+	}
+	else if (m_rightCorner)
+	{
+		m_tco->m_sampleBuffer->setRightFader(faderPos);
+	}
+	updateSample();
+}
+
+
+void SampleTCOView::mouseMoveLeftControlPoints(const QMouseEvent* e)
+{
+	float xpos = e->localPos().x();
+	float ypos = qMin(
+		qMax(e->localPos().y(), static_cast<qreal>(m_textLabelHeight)),
+		static_cast<qreal>(this->height())
+	);
+	float leftFaderAbsolutePos = m_tco->m_sampleBuffer->leftFaderPos() * this->width();
+	float Px = xpos / leftFaderAbsolutePos;
+	float Py = 1 - (ypos - m_textLabelHeight) / (this->height() - m_textLabelHeight);
+	if (m_moveLeftP1)
+	{
+		m_tco->m_sampleBuffer->setLeftP1(Px, Py);
+	}
+	else if (m_moveLeftP2)
+	{
+		m_tco->m_sampleBuffer->setLeftP2(Px, Py);
+	}
+	updateSample();
+}
+
+
+void SampleTCOView::mouseMoveRightControlPoints(const QMouseEvent* e)
+{
+	float xpos = e->localPos().x();
+	float ypos = e->localPos().y();
+	float rightFaderAbsolutePos = m_tco->m_sampleBuffer->rightFaderPos() * this->width();
+	float Px = (xpos - rightFaderAbsolutePos) / (this->width() - rightFaderAbsolutePos);
+	float Py = 1 - (ypos - m_textLabelHeight) / (this->height() - m_textLabelHeight);
+	if (m_moveRightP1)
+	{
+		m_tco->m_sampleBuffer->setRightP1(Px, Py);
+	}
+	else if (m_moveRightP2)
+	{
+		m_tco->m_sampleBuffer->setRightP2(Px, Py);
+	}
+	updateSample();
+}
 
 
+void SampleTCOView::mouseMoveEvent(QMouseEvent* e)
+{
+	if ((e->buttons() & Qt::LeftButton) && (m_leftCorner || m_rightCorner))
+	{
+		mouseMoveFadingPos(e);
+	}
+	else if ((e->buttons() & Qt::LeftButton) && (m_moveLeftP1 || m_moveLeftP2))
+	{
+		mouseMoveLeftControlPoints(e);
+	}
+	else if ((e->buttons() & Qt::LeftButton) && (m_moveRightP1 || m_moveRightP2))
+	{
+		mouseMoveRightControlPoints(e);
+	}
+	else
+	{
+		// This event is only forwarded, when the fades are not currently adjusted, in order
+		// to prevent the SampleTCO from beeing moved around in the timeline when adjusting
+		// the fades.
+		TrackContentObjectView::mouseMoveEvent(e);
+	}
+}
+
+
+void SampleTCOView::setSFade()
```

Comment:
I think for these `set*Fade` functions it wouldn't hurt to explain why those constants are used in the `set(Left|Right)P[12]` calls.

Reply:
Done.